server/protocols/jmap/src/test/java/org/apache/james/jmap/crypto/JamesSignatureHandlerTest.java

```diff
@@ -34,7 +42,26 @@
 
     @Before
     public void setUp() throws Exception {
-        signatureHandler = new JamesSignatureHandlerProvider().provide();
+       signatureHandler = new JamesSignatureHandlerProvider().provide();
+    }
+
+    private static class MockJamesSignatureHandler extends JamesSignatureHandler {
+        public MockJamesSignatureHandler(FileSystem fileSystem, JMAPConfiguration jmapConfiguration) {
+            super(fileSystem, jmapConfiguration);
+        }
+
+        @Override
+        @VisibleForTesting Certificate fetchAlias(KeyStore keystore) throws KeyStoreException {
+            return null;
```

Comment:
:-1: we could simply generate a keystore with the bad ALIAS: 

```
keytool -genkey -alias badAlias -keyalg RSA -keystore src/resources/keystore
```

That way we would avoid that uggly MockJamesSignatureHandler

Reply:
The `MockJamesSignatureHandler` is clearly ugly, but do you have any example of `keytool` usage? because it looks like it modify some shared values, which is worse.

Reply:
Sorry, you generate manually a keystore using keytool (calle `keystore_with_wrong_alias`) that you include in `src/resources`.

How do that is ugly and worse? 
Isn't it closer to the real life scenario? 
Doesn't it place the test at the API level? 
Doesn't it minimize the assumptions your tests does on running code?

Reply:
Sure, it'll be closer to our case, but my concern is about side effects and failure.

If you have a way to change the `KeyStore` reference back and forth safely, it's ok, but if not it may cause a huge number of cascading test failure.

Reply:
It's not conplicated! The JMAP configuration takes a `keystore` option that we can make point the classpath (using `classpath://wrongKeystore`)...

Reply:
It looked pretty complicated but I've took your best ideas, thanks, that's way better.